### PR TITLE
Chore: Convert TimelineHeaderRow to RTL

### LINKED ISSUE #51741: Convert `packages/jaeger-ui-components/src/TraceTimelineViewer/TimelineHeaderRow/TimelineHeaderRow.test.js` to RTL

### PR DESCRIPTION
**Pull request by Seyaji**
**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/51741

### CODE REVIEW

**Comment by joey-grafana** [comment]
Hey [Seyaji](https://github.com/Seyaji), thanks so much for your contribution :) I can see that some of the frontend tests are failing. Are you ok to update those?

**Comment by Seyaji** [comment]
@joey-grafana Apologies for not updating this I have been swamped recently and haven't had a chance to get back to it. I'll see what I can do tonight / this weekend if that's ok?

**Comment by Seyaji** [comment]
@ashharrison90  @joey-grafana  I have attempted to fix the failing TraceView test, Its strange that ``` data-testid[span-view]``` cannot be found as I didn't modify that in my original pull request, to debug I began to rewrite the tests one by one (they may be wrong 😅) . 

There is data for three spans but the Process switch doesn't seem to appear on user click. Maybe I am going about this the wrong way.  I would appreciate an extra set of eyes before I go further down the rabbit hole 🙏

**Comment by Clarity-89** [comment]
@Seyaji could you update your branch to the latest main? Looks like there are some merge conflicts.

File: packages/jaeger-ui-components/src/TraceTimelineViewer/TimelineRow.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ export function TimelineRowCell(props: TimelineRowCellProps) {
   const widthPercent = `${width * 100}%`;
   const mergedStyle = { ...style, flexBasis: widthPercent, maxWidth: widthPercent };
   return (
-    <div className={cx(ubRelative, className)} style={mergedStyle} {...(rest as any)}>
+    <div className={cx(ubRelative, className)} style={mergedStyle} {...(rest as any)} data-testid="TimelineRowCell">
```
Comment (Clarity-89) [inline packages/jaeger-ui-components/src/TraceTimelineViewer/TimelineRow.tsx:64]:
```suggestion
    <div className={cx(ubRelative, className)} style={mergedStyle} data-testid="TimelineRowCell" {...rest}>
```
It looks like by putting this testid after the rest props you're overriding `span-view` testid, which is assigned to the same component. In general we need to be careful which components we add testid to as it seems we assign different ids to the same components.

**Comment by CrypticSignal** [comment]
Hi @Seyaji 🙂

We're almost there! If you look at the details of the failing `continuous-integration/drone/pr` check, you can see that the [test-frontend step](https://drone.grafana.net/grafana/grafana/77585/1/9) is failing. If you look at the logs for that step, you will see where the issue is.

You need to change `data-test-id` in `public/app/features/explore/TraceView/TraceView.test.tsx` to `data-testid` 🙂

With regards to the [betterer-frontend step](https://drone.grafana.net/grafana/grafana/77585/1/8) failing, try re-merging your branch with the latest main branch. This will hopefully fix that.

**Review by ashharrison90 — APPROVED** [review]
i've pushed the selector fix as well

lgtm, thanks again @Seyaji  🙌

**Pull request merged**